pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -370,6 +371,12 @@ def _evaluate(
 
                 # track outputs for collation
                 if output is not None:
+
+                    # Add step predictions to prediction collection to write later
+                    do_write_predictions = is_result_obj and test_mode
+                    if do_write_predictions:
```

Comment:
1. this should just stay inside the prediction object
2. we already track prediction objects. No need to do anything explicit.